integration-tests/oauth2/src/main/java/com/okta/spring/tests/oauth2/opaquetoken/BasicOpaqueTokenResourceServerApplication.java

```diff
@@ -0,0 +1,112 @@
+ */
+package com.okta.spring.tests.oauth2.opaquetoken;
+
+import com.okta.spring.boot.oauth.Okta;
+import org.springframework.boot.SpringApplication;
+import org.springframework.boot.autoconfigure.SpringBootApplication;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpMethod;
+import org.springframework.security.access.prepost.PreAuthorize;
+import org.springframework.security.config.annotation.method.configuration.EnableGlobalMethodSecurity;
+import org.springframework.security.config.annotation.web.builders.HttpSecurity;
+import org.springframework.security.config.annotation.web.configuration.WebSecurityConfigurerAdapter;
+import org.springframework.security.core.Authentication;
+import org.springframework.web.bind.annotation.CrossOrigin;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.security.Principal;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.Map;
+
+@SpringBootApplication
+@EnableGlobalMethodSecurity(prePostEnabled = true, securedEnabled = true)
+public class BasicOpaqueTokenResourceServerApplication {
+
+    public static void main(String[] args) {
+        SpringApplication.run(BasicOpaqueTokenResourceServerApplication.class, args);
+    }
+
+    @Configuration
+    static class OpaqueTokenResourceSecurityConfigurer extends WebSecurityConfigurerAdapter {
+
+        @Override
+        public void configure(HttpSecurity http) throws Exception {
+
+            Okta.configureResourceServer401ResponseBody(http);
+
+            http.authorizeRequests()
+                .antMatchers(HttpMethod.OPTIONS,"/**").permitAll()
+                .anyRequest().authenticated()
+                .and().oauth2ResourceServer().opaqueToken();
+        }
+    }
+
+    @RestController
+    @CrossOrigin(origins = "http://localhost:8080")
+    public static class MessageOfTheDayController {
+
+        @GetMapping("/api/userProfile")
+        @PreAuthorize("hasAuthority('SCOPE_profile')")
+        public Map<String, Object> getUserDetails(Authentication authentication) {
+            // TODO: validate this is the correct way to get the details
+            return (Map<String, Object>) authentication.getDetails();
+        }
+
+        @GetMapping("/api/messages")
+        @PreAuthorize("hasAuthority('SCOPE_email')")
+        public Map<String, Object> messages() {
+
+            Map<String, Object> result = new HashMap<>();
+            result.put("messages", Arrays.asList(
+                    new Message("I am a robot."),
+                    new Message("Hello, world!")
+            ));
+
+            return result;
+        }
+
+        @GetMapping("/api/invalidScope")
+        @PreAuthorize("hasAuthority('SCOPE_invalid')")
+        public String invalidScope() {
+            throw new IllegalStateException("Test exception, user should not have access to this method");
```

Comment:
You may want to explicitly return a `500` here instead, so it's easier to track down failures

Reply:
Do we really want to categorize it as a `500` error?

Moreover, I used this to be consistent with a similar use case (for groups though) at https://github.com/okta/okta-spring-boot/blob/2da169d7f9d6674504d15b3c41ab0a4e5a0506a0/integration-tests/oauth2-reactive/src/main/java/com/okta/spring/tests/oauth2/reactive/code/ReactiveRedirectCodeFlowApplication.java#L49-L53.

Reply:
@bdemers thoughts on this one?

Reply:
Either option is fine. Throwing the llegalStateException will cause a 500 indirectly anyway